### PR TITLE
[JENKINS-58732] Update dependencies to account for moved class

### PR DESCRIPTION
**Pull request by Vlatombe**
# Description

Considering the dependency tree, I had to bump quite a few to get this compiling...

See [JENKINS-58732](https://issues.jenkins-ci.org/browse/JENKINS-58732).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by dwnusbaum** [comment]
The changes look fine to me, but there are a lot more ATH failures than normal, so I'm trying a rebuild.

**Comment by dwnusbaum** [comment]
The same ATH tests failed in the rebuild, so IDK. The only interesting error I noticed in any of the failed test logs was:

```
15:22:14 INFO  - 1585063315097 - SEVERE - http://10.128.0.56:58293/blue/rest/search/?q=type:pipeline;organization:undefined;pipeline:;excludedFromFlattening:jenkins.branch.MultiBranchProject,hudson.matrix.MatrixProject&filter=no-folders&start=0&limit=26 - Failed to load resource: the server responded with a status of 400 (Bad Request)
```

That came from [this failure](https://ci.blueocean.io/job/blueocean/job/PR-2056/4/testReport/io.blueocean.ath.offline.dashboard/DashboardTest/ATH___Jenkins_2_138_4___checkExecutorPluginLoaded/). About half of the screenshots for the ATH failures are blank images, and the rest look like a page failed to load.

Maybe some of the updated plugins are failing to load in the ATH because it is running against 2.138.x?

**Comment by Vlatombe** [comment]
> Maybe some of the updated plugins are failing to load in the ATH because it is running against 2.138.x?

I'll check. But it sounds like the kind of thing that should be checked through an enforcer rule.

**Comment by olamy** [comment]
Need deeper investigation as ATH results contains such error
```
Stacktrace
org.apache.http.client.HttpResponseException: Not Found
```

**Comment by olamy** [comment]
@Vlatombe cherry pick this commit e25b2bf48ce4ae01db1d622393a081422b98a8ef and we will be fine

**Comment by Vlatombe** [comment]
@olamy Done. Thank you for looking it up!

**Comment by Vlatombe** [comment]
2 remaining failures, similar to failures observed on master (https://ci.blueocean.io/job/blueocean/job/master/1876/) so I guess they are flaky.

File: acceptance-tests/runner/runtime-plugins/runtime-deps/pom.xml [hunk 1]
```diff
@@ -69,14 +69,19 @@
...
         <!-- following are required to run on newer jenkins versions (after 2.7) as it changed how classloading works a bit for plugins -->
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>structs</artifactId>
-            <version>1.17</version>
+            <version>1.20</version>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>apache-httpcomponents-client-4-api</artifactId>
```
Comment (jglick) [inline acceptance-tests/runner/runtime-plugins/runtime-deps/pom.xml]: `dependencyManagement`?

Comment (olamy) [reply acceptance-tests/runner/runtime-plugins/runtime-deps/pom.xml]: would be better but this is only use for BO testing this dependency cannot be used by any project. So I have been lazy :)

**Pull request merged**